internal/pkg/skuba/deployments/ssh/files.go

```diff
@@ -20,24 +20,48 @@ package ssh
 import (
 	"encoding/base64"
 	"fmt"
+	"os"
 	"path"
 
 	"k8s.io/klog"
 )
 
 // UploadFileContents creates a file with the content sent
 // into a target system's specific path.
-func (t *Target) UploadFileContents(targetPath, contents string) error {
+func (t *Target) UploadFileContents(targetPath, contents string, perm os.FileMode) error {
 	klog.V(1).Infof("uploading to remote file %q with contents", targetPath)
 	dir, _ := path.Split(targetPath)
 	encodedContents := base64.StdEncoding.EncodeToString([]byte(contents))
 	if _, _, err := t.silentSsh("mkdir", "-p", dir); err != nil {
 		return err
 	}
-	_, _, err := t.silentSshWithStdin(encodedContents, "base64", "-d", "-w0", fmt.Sprintf("> %s", targetPath))
+	_, _, err := t.silentSshWithStdin(encodedContents, "umask", osFileModeToUmaskSymbolMode(perm), "&&", "base64", "-d", "-w0", fmt.Sprintf("> %s", targetPath))
```

Comment:
We can avoid the inverse calculation of the umask if we do `install -m <mode> /dev/null <target> && base64...` to create the file -- it also reinforces the mode if the file exists already.

Reply:
> We can avoid the inverse calculation of the umask if we do `install -m <mode> /dev/null <target> && base64...` to create the file -- it also reinforces the mode if the file exists already.

Sound good, I had thought of using `install` before, let me update the PR.